torch/csrc/jit/codegen/cuda/lower_index.cpp

```diff
@@ -270,6 +270,150 @@ void IndexLowering::visit(const kir::ReductionOp* rop) {
   }
 }
 
+namespace {
+
+template <typename T>
+kir::Allocate* allocGlobalBuffer(
```

Comment:
Can we use this to also simplify the grid reduction code? Would make more sense to do in a follow up if yes.

Reply:
Yes I think so. Thanks for pointing this out. I will put further simplifications in a follow up.